pwiz_tools/Skyline/TestFunctional/AuditLogValidationTest.cs

```diff
@@ -0,0 +1,105 @@
+            // - valid audit log file - opens cleanly
+            var documentFile = TestFilesDir.GetTestPath(@"AuditLogValidationTest/MethodEditClean.sky");
+            WaitForCondition(() => File.Exists(documentFile));
+
+            //checking if the library files exist where we expect them.
+            WaitForCondition(() => File.Exists(TestFilesDir.GetTestPath(@"AuditLogValidationTest/yeast_cmp_20.hlf")));
+            WaitForCondition(() => File.Exists(TestFilesDir.GetTestPath(@"AuditLogValidationTest/yeast_cmp_20.slc")));
+
+            try
+            {
+                ShowDialog<MessageDlg>(() => SkylineWindow.OpenFile(documentFile), 200);
+                Assert.Fail("Failed to open a well-formed document with normal audit log.");
+            }
+            catch (AssertFailedException e)
+            {
+                //nothing to do, this is an expected outcome of the normal test run.
+            }
+
+            // - audit log with modified content - expect message dialog to appear
+            // - audit log with modified entry hashes - expect exception
+            foreach(string fileName in new[] { @"AuditLogValidationTest/MethodEditContentMod.sky", @"AuditLogValidationTest/MethodEditEntryHashMod.sky" })
+            {
+                TestInvalidFile(fileName, AuditLogStrings
+                    .AuditLogList_ReadFromFile_The_following_audit_log_entries_were_modified);
+            }
+
+            // - audit log with entry timestamps out of order - expect exception
+            TestInvalidFile(@"AuditLogValidationTest/MethodEditEntryOrderMod.sky",
+                AuditLogStrings
+                    .AuditLogList_Validate_Audit_log_is_corrupted__Audit_log_entry_time_stamps_and_indices_should_be_decreasing);
+
+        }
+
+        private void TestInvalidFile(string fileName, string expectedMessage)
+        {
+            //replace print formats with regex and compile the testing regex.
+            expectedMessage = new Regex(@"\{[0-9]+\}").Replace(expectedMessage, @"[0-9]+");
+            expectedMessage = new Regex(@"\{[0-9]+:G\}").Replace(expectedMessage, ".*");
+            Regex reTest = new Regex(expectedMessage);
+
+            var documentFile = TestFilesDir.GetTestPath(fileName);
+            //if no dialog is opened the method will fail on timeout
+            var messageDialog = ShowDialog<MessageDlg>(() => SkylineWindow.OpenFile(documentFile), 200);
+            //making sure we get the expected error message
+            string dialogMessage = "(none)";
+            RunUI(() => dialogMessage = messageDialog.Message);
+            if (reTest.Match(dialogMessage).Value == String.Empty)
+            {
+                Console.Write("Error message: " + dialogMessage);
+                Assert.Fail("Unexpected exception type when opening document with entries out of order audit log.");
+            }
+            //close the dialog
+            RunUI(() => messageDialog.ClickOk());
```

Comment:
It would probably be better to do:
OkDialog(messageDialog, messageDialog.ClickOk)
since that will make sure that the test waits for the dialog to disappear.